### PR TITLE
[Fix][Transaction] Fix transaction can not be enable when allowAutoTopicCreation = false

### PR DESCRIPTION
**Pull request by liangyepianzhou**
Fix https://github.com/apache/pulsar/issues/14786
### Motivation
Fix transaction can not be enable when allowAutoTopicCreation = false
### Modification
create topic, no only metadata when init transaction

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
How about the transaction log topic and transaction buffer snapshot topic? they will create allow auto-creation or not?

**Comment by liangyepianzhou** [comment]
> How about the transaction log topic and transaction buffer snapshot topic? they will create allow auto-creation or not?
Transaction Log is to open a manager Ledger directly, and Transaction buffer can be created automatically.

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/transaction/TransactionTest.java [hunk 1]
```diff
@@ -159,4 +165,17 @@ public void transferNormalTest(Supplier<String> serviceUrl) throws Exception {
...
+    @Test(dataProvider = "ServiceAndAdminUrls")
+    public void testAutoCreatePartitionTopicOfTC(Supplier<String> serviceUrl, Supplier<String> adminUrl) throws Exception {
+        log.info("create transaction coordinator test start.");
+
+        @Cleanup
+        PulsarAdmin admin = PulsarAdmin.builder().serviceHttpUrl(adminUrl.get()).build();
+
+        Awaitility.await().untilAsserted(() ->
+                Assert.assertTrue(admin.topics().getList(TopicName.TRANSACTION_COORDINATOR_ASSIGN.getNamespace())
+                .contains(TopicName.TRANSACTION_COORDINATOR_ASSIGN.toString())));
```
Comment (codelipenghui) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/transaction/TransactionTest.java:177]: Will fail here if the coordinator assign topic is a partitioned topic.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 2]
```diff
@@ -2836,7 +2835,7 @@ public boolean isAllowAutoTopicCreation(final String topic) {
 
     public boolean isAllowAutoTopicCreation(final TopicName topicName) {
         //System topic can always be created automatically
-        if (pulsar.getConfiguration().isSystemTopicEnabled() && checkTopicIsEventsNames(topicName)) {
+        if (pulsar.getConfiguration().isSystemTopicEnabled() && isSystemTopic(topicName)) {
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:2838]: Do we need to check the system topic is enabled?

Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:2838]: Why not?

**Pull request closed without merging**